Dockerfile

```diff
@@ -58,10 +64,37 @@ RUN jupyter labextension install nbdime-jupyterlab --no-build && \
         jupyter lab clean && \
         jlpm cache clean && \
         npm cache clean --force && \
-        rm -rf $HOME/.node-gyp && \
-        rm -rf $HOME/.local && \
-    fix-permissions $CONDA_DIR $HOME
+        rm -rf "${HOME}/.node-gyp/" && \
+        rm -rf "${HOME}/.local/" && \
+    fix-permissions "${CONDA_DIR}" "${HOME}"
+
+# READ-ONLY notebooks
+RUN mkdir -p "notebooks"
+COPY notebooks/* notebooks/
+RUN jupyter trust notebooks/*ipynb
+
+# Dont want it called work as well as ONED
+# to avoid confusion. But dont remove if its got anything in it.
+RUN rmdir work
+
+# generate non-volative disk
+RUN mkdir -p notebooks/data && fix-permissions notebooks/data
+
+###########################################
+# This might be repeated in derived dockers
+ENV COURSENAME ${CONDA_ENV}
+# sort local directories
+# local mount point for OneDrive: it should reside in home
+ARG ONED='OneDrive - University College London'
+ENV ONEDRIVE "${HOME}/${ONED}"
+# for mounting: in the home directory
+RUN mkdir -p "${ONEDRIVE}/${COURSENAME}"
+VOLUME "${ONEDRIVE}/${COURSENAME}"
+
+# make a link in case base of jupyter in notebooks
+#RUN mkdir -p "notebooks/${ONED}"
+RUN ln -s "${ONEDRIVE}/${COURSENAME}" "notebooks/${COURSENAME}"
+RUN mv notebooks/README.md /tmp && sed < /tmp/README.md 's/__COURSE__/'"${COURSENAME}"'/g' > notebooks/README.md
```

Comment:
Other possibility might be having a `course_name.conf` file in the main folder, so that the README doesn't need to be tailored to this.